spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/feign/TraceLoadBalancerFeignClient.java

```diff
@@ -49,7 +49,7 @@
 	HttpTracing httpTracing;
 	TracingFeignClient tracingFeignClient;
 
-	TraceLoadBalancerFeignClient(Client delegate,
+	public TraceLoadBalancerFeignClient(Client delegate,
```

Comment:
ex I mean to say if you provide a bean of subtype CachingSpringLoadBalancerFactory, your custom one.. why do you also need to expose this (the consumer of it)?

Reply:
Because the `CachingSpringLoadBalancerFactory` is dead by `@Primary`, the user can't use the custom `CachingSpringLoadBalancerFactory` externally unless the `LoadBalancerFeignClient` is overridden. And I also verified that you can use the custom `CachingSpringLoadBalancerFactory` after rewriting `LoadBalancerFeignClient`, which is all right. But when I reference sleuth in my project, `TraceLoadBalancerFeignClient` will replace my custom `LoadBalancerFeignClient` in the `TraceFeignObjectWrapper.wrap` method, unless my `LoadBalancerFeignClient` instance is inherited from `TraceLoadBalancerFeignClient`. You can look at this example https://github.com/spring-cloud/spring-cloud-openfeign/pull/50#issue-204580240